device/isis/routeradvanced.yaml

```diff
@@ -31,7 +32,6 @@ components:
             The rate at which LSPs are re-sent in seconds.
           type: integer
           minimum: 1
-          maximum: 294967295
```

Comment:
What is a sensible max that our users and engine implementors will actually test for?  I don't think leaving it open is a good idea, the max being 294967295 is also not a good sensible max (that's 9+ years for the refresh rate)?

Reply:
294967295 -- this is somebody missed out first digit in "4" of max int 4294967296. But corrected in the latest model. pls check.

Reply:
my question remains.  is 4294967296 a sensible max value?  sure that is the theoratical max for an int, but that is 49710 days -> 136 years.  Who is going to set it to that?

even for the 294967295 max value that was incorrectly defined was 9+ years.  What are sensible max value?

Reply:
seems like we feel it is okay to let people to put in value that probably no one realistically will ever run with.  why?

Reply:
i see apratim's response.  however, i think we should have sensible max.  if we put it as infinite meaning the max int value, is that documented that it is to mean infinite (no refreshes ever)?  will any real user test that scenario?  will we test for that scenario?

again, now, what happens to maxint-1, how will anyone test for that value?  maybe check with users what is the max they can think of for the refresh interval, and really limit it to that, and that value +1 being infinite if there is even a use case for that.  or maybe "0" should mean infinity to apratim's use case.

Reply:
Do you have any max-limit in mind?

Reply:
few references from cisco and frr.

https://docs.frrouting.org/en/latest/isisd.html
https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/iproute_isis/configuration/xe-3s/irs-xe-3s-book/irs-fscnt.html

their max is definitely less than a day, not like 136 years like we're leaving open because of maxint.

Reply:
another data point from https://github.com/openconfig/public/blob/master/release/models/isis/openconfig-isis.yang, which leads to similar max described on cisco/cumulus links above.  please leverage these open resources when deciding sensible max/min values.

    leaf lsp-refresh-interval {
      type uint16;
      units seconds;
      description
        "Time interval in seconds that specifies how often route topology
         that a device originates is transmitted in LSPs.";
    }
  }

Reply:
I think it is much less than a day.  It is probably no more than a few mins at the most.  The higher the interval the longer it takes for the protocol to converge.  Operators want network to converge much much faster than a few mins.  How about we keep it to 600 seconds like what seems to be suggested in the frr docs?  It will reduce our test surface and be somewhat reasonable to test.

Reply:
"lsp-refresh-interval" in above reference mentioned data type, without mentioning max and min limits of it. We will correct limit to max unit16.  This is value that lsp-refresh-interval and a user can configure with any value in [0, max_unit16]; if a user does not configure explicitly then the default value  is taken.
@ankur-sheth, are you suggesting the default to value to "600" instead of "900"?